test/unit/vdom/properties.js

```diff
@@ -74,6 +75,22 @@ describe('vdom/properties', () => {
     }).to.not.throw(Error);
   });
 
+  it('should set props as properties instead of attributes on ctor references', () => {
+    let fixture = document.createElement('div');
+    const Elem = define('x-test', {
+      props: {
+        foo: {}
+      }
+    });
+
+    IncrementalDOM.patch(fixture, () => {
+      vdom.element(Elem, { foo: 'bar' });
```

Comment:
I'd throw in another attribute here that *isn't* declared as a prop, and then assert that it *wasn't* set as a prop, assuming that's the desired behaviour.